clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/ApplicationCachingAgent.java

```diff
@@ -71,7 +75,42 @@ public String getAgentType() {
 
   @Override
   protected List<Application> getItems(AmazonECS ecs, ProviderCache providerCache) {
+    // get all ECS services
+    ServiceCacheClient serviceCacheClient = new ServiceCacheClient(providerCache, objectMapper);
+    Collection<Service> services = serviceCacheClient.getAll();
+    log.info("Found {} ECS services for which to cache applications", services.size());
+
+    Map<String, Map<String, Collection<String>>> appRelationships = new HashMap<>();
+
+    for (Service service : services) {
+      String applicationKey = service.getApplicationName();
```

Comment:
The logic for determining applications is the same as it exists today. The only difference being that we're performing this logic and writing it to the cache rather than performing it at query runtime.